src/Corax/Querying/MultiTermMatches/IndexSearcher.MultiTermMatches.Ranges.cs

```diff
@@ -170,12 +169,11 @@ private MultiTermMatch RangeBuilder<TLow, THigh>(FieldMetadata field, long low,
         where TLow : struct, Range.Marker
         where THigh : struct, Range.Marker
     {
-        var terms = _fieldsTree?.CompactTreeFor(field.FieldName);
-        if (terms == null)
+        if (_fieldsTree == null || _fieldsTree.TryGetCompactTreeFor(field.FieldName, out _) == false)
             return MultiTermMatch.CreateEmpty(_transaction.Allocator);
 
         field = field.GetNumericFieldMetadata<long>(Allocator);
-        var set = _fieldsTree?.LookupFor<Int64LookupKey>(field.FieldName);
+        var set = _fieldsTree.LookupFor<Int64LookupKey>(field.FieldName);
```

Comment:
This is a long value range builder, and it seems we're acquiring the compact tree only to check if there are any values. Can’t we perform the check directly using numerical lookups instead?

Reply:
Good question. But, shouldn't that happen at the query builder stage? I would not include that into this PR as that is entirely unrelated with the types of optimizations types of this PR. Check out if it can be done at the query builder itself and create an issue for it.

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-22951/IndexSearcher.MultiTermMatches.Ranges-building-optimization